### PR TITLE
[R2] Add unimplemented S3 header to CopyObject

### LINKED ISSUE #4508: S3 API compatibility: x-amz-checksum-algorithm
### Existing documentation URL(s)

- https://developers.cloudflare.com/r2/platform/s3-compatibility/api/

### Section that requires update

`CopyObject` section

### What needs to change?

Add `❌ x-amz-checksum-algorithm`

### How should it change?

The header should be added to denote that it's not currently supported.

Or, if some values of the header are supported and others aren't, document it to make that clear.

### Additional information

When I try to copy an object using this header, I get the error response:

```
S3: CopyObject, https response error StatusCode: 501, RequestID: , HostID: , api error NotImplemented: Header 'x-amz-checksum-algorithm' with value 'SHA256' not implemented
```

The [S3 docs](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html#) say:

> **Checksums**
> 
> When copying an object, if it has a checksum, that checksum will be copied to the new object by default. When you copy the object over, you may optionally specify a different checksum algorithm to use with the x-amz-checksum-algorithm header.

### PR DESCRIPTION
**Pull request by KianNH**
Closes https://github.com/cloudflare/cloudflare-docs/issues/4508

### CODE REVIEW

File: content/r2/platform/s3-compatibility/api.md [hunk 1]
```diff
@@ -61,25 +61,25 @@ Below is a list of unimplemented bucket-level operations.
...
...-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (deadlypants1973) [inline content/r2/platform/s3-compatibility/api.md]: Can we clean this up?

Comment (KianNH) [reply content/r2/platform/s3-compatibility/api.md]: Oops - not sure what happened there. I think my IDE tries to make tables pretty in code haha

Comment (KianNH) [reply content/r2/platform/s3-compatibility/api.md]: Fixed

Comment (deadlypants1973) [reply content/r2/platform/s3-compatibility/api.md]: I had to add another thing so I ended up making a new pr anyway but stole your checksum text, thank you :)

Comment (KianNH) [reply content/r2/platform/s3-compatibility/api.md]: Yoink

**Pull request closed without merging**